src/array.jl

```diff
@@ -173,24 +163,22 @@ CategoricalMatrix{T}(m::Int, n::Int; ordered=false) where {T} =
 # so that ordered!() does not affect the original array
 function CategoricalArray{T, N, R}(A::CategoricalArray{S, N, Q};
                                    ordered=_isordered(A)) where {S, T, N, Q, R}
-    U = unwrap_catvalue_type(T)
-    res = convert(CategoricalArray{U, N, R}, A)
+    V = unwrap_catvalue_type(T)
+    res = convert(CategoricalArray{V, N, R}, A)
     if res.pool === A.pool # convert() only makes a copy when necessary
-        res = CategoricalArray{U, N, R}(res.refs, deepcopy(res.pool))
+        res = CategoricalArray{V, N}(res.refs, deepcopy(res.pool))
```

Comment:
Why remove R? That was more explicit. Same below.

Reply:
Because it would require introducing `reftype(CategoricalPool)` that you wanted to remove.
`R` is already passed into inner ctor by `refs` and by `pool`.
Also in many cases inner ctor usage `R` was not explicitly available, so it was extracted from the `refs`.
This scheme seemed a little bit redundant for me, also the removal of `R` seemed to be in line with simplifying the `CategoricalPool` ctor params.

Reply:
In the present case `R` is available so better pass it. That also makes it certain and clear that `res` doesn't change type depending on whether pools are equal, which would be a serious problem.

Reply:
There's only want ctor that accepts `CategoricalPool` as the parameter: the inner one.
So all these cases with `R` removal are referring to it.

I can introduce a "low-level" outer ctor with exactly the same signature as the inner one + `R` in the type parameters list to make the cases like the one here this more explicit and do a triple `R` check.

Reply:
Ah, sorry, now I realize that you removed the `R` parameter to the inner constructor, which explains everything.

Reply:
So it's fine to keep it as it is now, or you prefer that I implement my suggestion from the comment above?

Reply:
No, let's say it's fine as-is, we have enough type parameters floating around already.